src/schema/objects/metadata.yaml

```
@@ -2986,7 +2986,7 @@ TaskName:
     Name of the task.
     No two tasks should have the same name.
     The task label included in the file name is derived from this `"TaskName"` field
-    by removing all non-alphanumeric (`[a-zA-Z0-9]`) characters.
+    by removing all non-alphanumeric (`[^0-9a-zA-Z]`) characters.
```

Perhaps this would be a better clarification?

```suggestion
    by removing all non-alphanumeric characters (that is, all except those matching `[0-9a-zA-Z]`).
```